Shopify/checkout-sheet-kit-swift
# Prevent retaining script handler / webview

### What changes are you making?

We are retaining `WKScriptMessageHandler`s in some cases (e.g. when a checkout completes, and is closed). As our script handler is `CheckoutWebView` this is causing the WebView itself to be retained, and leak.

This adds a wrapper class which uses a weak ref to the CheckoutWebView, allowing it to be cleared up. Based on this:
https://stackoverflow.com/questions/26383031/wkwebview-causes-my-view-controller-to-leak/26383032#26383032

### How to test

On main:

1. Run the sample app.
2. Open checkout
3. Check in Safari dev tools for the # of inspectable WebViews (it should be 1)
4. Complete checkout
5. Close checkout 
6. Begin a new checkout
7. Check in Safari dev tools for the # of inspectable WebViews (it should be 1, but is 2)

Repeat on this branch, and check that the count is now 1 in step 7.

### Before you merge

> [!IMPORTANT]
> - [ ] I've added tests to support my implementation
> - [x] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CONTRIBUTING.md).
> - [x] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CODE_OF_CONDUCT.md).
> - [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-swift).

---

<details>
<summary>Checklist for releasing a new version</summary>

- [ ] I have added a [Changelog](./CHANGELOG.md) entry.
</details>

> [!TIP]
> See the [Contributing documentation](./CONTRIBUTING.md#releasing-a-new-version) for instructions on how to publish a new version of the library.
